e2etest/pkg/frr/bgp.go

```diff
@@ -124,3 +124,27 @@ func ContainsCommunity(exec executor.Executor, community string) error {
 	}
 	return nil
 }
+
+func allroutesMatchLocalPref(routes map[string]bgpfrr.Route, localPref uint32) error {
+	for _, route := range routes {
+		if route.LocalPref != 0 && route.LocalPref != localPref {
```

Comment:
Doesn't this risk to have false positives?

Reply:
what do u mean ? localPref 0 check to filter out ebGP session since its not set for them ?

Reply:
What I mean is, if we set a localPref from outside, and all the routes have LocalPref = 0 this will pass even if it's wrong.

Reply:
routes will hjave localpref of 0 only for eBGP and for those we need to skip the test. iBGP if u don't set it its default to 100

Reply:
> routes will hjave localpref of 0 only for eBGP and for those we need to skip the test. iBGP if u don't set it its default to 100

Ok, can we then skip the test for eBGP routes / container and remove the != 0 so we are more explicit in what we are testing?

Reply:
is there a way in bgp_test to specify ? I coouldn't find away yesterday ?

Reply:
You are testing against both external frr containers. Picking the ebgp one should make the trick I think

Reply:
they are both called frrContainers I am using containersNetwork check that should let me run IBGP only

Reply:
you can filter by name, or have a reference to the ebgp one.